modules/hpe_nimble_volume.py

```diff
@@ -0,0 +1,897 @@
+    volcoll:
+        required: False
+        type: str
+        default: null
+        description:
+        - Name of volume collection of which this volume is a member.
+    force:
+        required: False
+        type: bool
+        default: False
+        description:
+        - Forcibly offline, reduce size or change read-only status a volume.
+    force_vvol:
+        required: False
+        type: bool
+        default: False
+        description:
+        - Forcibly move a Virtual Volume.
+    move:
+        required: False
+        type: bool
+        default: False
+        description:
+        - Move a Volume to different Pool.
+
+extends_documentation_fragment: hpe_nimble
+short_description: "Manages a HPE Nimble Storage volume"
+version_added: "2.9"
+'''
+
+EXAMPLES = r'''
+
+      # if state is create, then create a volume if not present. Fails if already present.
+      # if state is "present", then create a volume if not present, But Success if it already exists
+    - name: Create Volume if not present
+      hpe_nimble_volume:
+        hostname: "{{ hostname }}"
+        username: "{{ username }}"
+        password: "{{ password }}"
+        state: "{{ state | default('present') }}" # fail if exist
+        size: "{{ size }}"
+        limit_iops: "{{ limit_iops }}"
+        limit_mbps: 5000
+        force: false
+        metadata: "{{ metadata }}"
+        description: "{{ description }}"
+        name: "{{ name }}"
+
+    - name: Changing Volume "{{ name }}" to offline state
+      hpe_nimble_volume:
+        hostname: "{{ hostname }}"
+        username: "{{ username }}"
+        password: "{{ password }}"
+        state: offline
+        name: "{{ name }}"
+
+    - name: Changing Volume "{{ name }}" to online state
+      hpe_nimble_volume:
+        hostname: "{{ hostname }}"
+        username: "{{ username }}"
+        password: "{{ password }}"
+        state: online
+        name: "{{ name }}"
+
+    # Create a clone from the given snapshot name.
+    # if snapshot name is not provided then create a snapshot of source volume
+    # clone task should only run if parent_volume is specified.snapshot name is optional
+    - name: Let's create or refresh a clone!
+      hpe_nimble_volume:
+        hostname: "{{ hostname }}"
+        username: "{{ username }}"
+        password: "{{ password }}"
+        name: "{{ name }}" # name here is the name of cloned volume
+        parent_volume: "{{ parent_volume | mandatory }}"
+        snapshot: "{{ snapshot | default(None)}}"
+        state: "{{ state | default('present') }}" # fail if exist
+      when:
+        - parent_volume is defined
+
+    - name: Let's destroy myvol1 (it's not offline)
+      hpe_nimble_volume:
+        name: "{{ name }}"
+        state: absent
+
+    # if no snapshot is given then restore volume to last snapshot, fails if no snapshots exist
+    # or if snapshot is provided then restore volume from that snapshot
```

Comment:
# If no snapshot is given, then restore volume to last snapshot. Fails if no snapshots exist.
    # If snapshot is provided, then restore volume from specified snapshot.

Reply:
done